### DISCUSSION TITLE
Can you launch Sleek from the terminal and set an active todo.txt?

### DISCUSSION THREAD

**Discussion by gal064** [ransome1/sleek#596] [Q&A]
I'm using Joplin for note taking and it has this feature where you can open a file in an external editor. You need to choose the editor executable and the file is passed as an argument

I want to be able to choose a note that has todo.txt format and open it with Sleek. I could just run a shell script as the executable, and then open Sleek and pass the file URL. Is that possible or is there another workflow to allow me to open a todo.txt with Sleek?

**Comment by ransome1** [comment]
Hi @gal064.
> I want to be able to choose a note that has todo.txt format and open it with Sleek.

Where do you want to choose it? In terminal, or in the context menu of your file explorer in Joplin. Maybe you can walk me through your use case one more time with as many details as possible.

**Reply by gal064** [reply]
I'm flexible.

Basically what Joplin, and most apps that allow you to open files in external editors, does is that it calls the program and adds the path to the selected file as the first argument. 

That's a typical format for most editors out there ([Typora](https://support.typora.io/Use-Typora-From-Shell-or-cmd/), [VSCode](https://code.visualstudio.com/docs/editor/command-line)) where they take the first argument as a path to a file to open.

However, personally I can be more flexible. For example it looks like Sleek use to have an option to set the [opened file](https://github.com/ransome1/sleek/issues/173) using `SLEEK_CUSTOM_FILE` env variable. If that was available, I could also just create a shell script that Joplin loads. The shell script will take the first argument, set it into the env and then open Sleek.
